src/components/MonthSlider.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  CalendarDaysIcon,
+  ChevronLeftIcon,
+  ChevronRightIcon,
+} from '@heroicons/react/20/solid'
+import { Dispatch, SetStateAction, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Link, useNavigate, useSearchParams } from 'react-router-dom'
+
+import {
+  dateFromMonthYear,
+  monthYearFromDate,
+  shortTranslatedMonthFormat,
+  translatedMonthFormat,
+} from '../lib/dates'
+import isSupported from '../lib/is-supported'
+import { Budget, Translation } from '../types'
+import MonthPicker from './MonthPicker'
+
+type Props = {
+  budget: Budget
+  route?: string
+  setIsLoading: Dispatch<SetStateAction<boolean>>
+}
+
+const previousMonth = (yearMonth: string) => {
+  const [year, month] = yearMonth.split('-')
+
+  if (month === '01') {
+    return new Date(parseInt(year) - 1, 11, 15)
+  } else {
+    return new Date(parseInt(year), parseInt(month) - 2, 15)
+  }
+}
+
+const nextMonth = (yearMonth: string) => {
+  const [year, month] = yearMonth.split('-')
+
+  if (month === '12') {
+    return new Date(parseInt(year) + 1, 0, 15)
+  } else {
+    return new Date(parseInt(year), parseInt(month), 15)
+  }
+}
+
+const MonthSlider = ({ budget, route = '', setIsLoading }: Props) => {
+  const { t }: Translation = useTranslation()
+  const navigate = useNavigate()
+
+  const [searchParams] = useSearchParams()
+  const activeMonth =
+    searchParams.get('month')?.substring(0, 7) || monthYearFromDate(new Date())
```

Comment:
I think it makes more sense to pass this in the props. Depending on the usecase, we might not always encode the month in the url, and even if we do it doesn't make sense to parse it both in the `Dashboard.tsx` and here.

Reply:
Thanks for the feedback, I changed it in https://github.com/envelope-zero/frontend/pull/1518/commits/e7742d91d16ba5a5bd33150be89f403ba8c291a5. Is this what you had in mind?